src/CoWSwapEthFlow.sol

```diff
@@ -35,16 +36,35 @@ contract CoWSwapEthFlow is
     /// onchain data.
     mapping(bytes32 => EthFlowOrder.OnchainData) public orders;
 
-    /// @param _cowSwapSettlement The address of the CoW Swap settlement contract.
-    /// @param _wrappedNativeToken The address of the default native token in the current chain (e.g., WETH on mainnet).
+    /// @param _cowSwapSettlement The CoW Swap settlement contract.
+    /// @param _wrappedNativeToken The default native token in the current chain (e.g., WETH on mainnet).
     constructor(
         ICoWSwapSettlement _cowSwapSettlement,
-        IERC20 _wrappedNativeToken
+        IWrappedNativeToken _wrappedNativeToken
     ) CoWSwapOnchainOrders(address(_cowSwapSettlement)) {
         cowSwapSettlement = _cowSwapSettlement;
         wrappedNativeToken = _wrappedNativeToken;
+
+        _wrappedNativeToken.approve(
```

Comment:
will this always happen even if the user doesn't need to approve?

Reply:
There always has to be an approval to trade with the settlement contract. However, this contract only sends an approval once when it's deployed and doesn't need to happen for each user trade.